src/widgets/block.rs

```diff
@@ -703,13 +721,24 @@ impl<'a> Block<'a> {
     }
 }
 
-impl<'a> Widget for Block<'a> {
-    fn render(self, area: Rect, buf: &mut Buffer) {
-        if area.area() == 0 {
-            return;
+/// An extension trait for [`Block`] that provides some convenience methods.
+///
+/// This is pub(crate) for now because it is not clear if this is the best way to do this.
+pub(crate) trait BlockExt {
```

Comment:
This seems like a good idea for now at least!

Reply:
I'm not 100% sure I like the render and modify Rect being tied always. that could be annoying for a caller that wants to do something with the original area. (Example: put a scrollbar on the block border instead of within the border).

Making this pub(crate) makes it easy to change this without breaking user code.

Reply:
Maybe the name `render` should just be changed to `render_option` or similar. To avoid confusion and be able to switch render easily.

Reply:
That could work. Maybe we could also add the following usefully:

```rust
fn inner() -> Rect { ... }
```

and split the render call into two (this kinda looks a little oddish):
```rust
self.block.render_optional(area, buf);
let area = self.block.inner_optional();
```

Reply:
> ```rust
> self.block.render_optional(area, buf);
> let area = self.block.inner_optional();
> ```

I didn't like this approach at first but changing `_optional` to`_if_exists` feels more descriptive to me.

```rust
self.block.render_if_exists(area, buf);
let area = self.block.inner_if_exists();
```

Reply:
I decided to split this and leave it as render(), it seems to look better with two calls (render, inner) rather than one that mutates the area.